src/platform/linux/midiout_device_alsa.cpp

```diff
@@ -0,0 +1,199 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "midiout_device_alsa.h"
+#include "output.h"
+#include "system.h"
+
+AlsaMidiOutDevice::AlsaMidiOutDevice() {
+	int status = snd_seq_open(&midi_out, "default", SND_SEQ_OPEN_DUPLEX, 0);
+	if (status < 0) {
+		Output::Debug("ALSA: snd_seq_open failed: {}", snd_strerror(status));
+		return;
+	}
+
+	snd_seq_client_info_t* client_info;
+	snd_seq_port_info_t* port_info;
+	snd_seq_client_info_alloca(&client_info);
+	snd_seq_port_info_alloca(&port_info);
+
+	// TODO: This simply enumerates all devices and connects to the last matching one
+	// There should be a way to configure this
+	// The last is usually timidity++ or fluidsynth, so this works
+	std::string dst_client_name;
+	std::string dst_port_name;
+	bool candidate_found = false;
+
+	snd_seq_client_info_set_client(client_info, -1);
+	while (snd_seq_query_next_client(midi_out, client_info) == 0) {
+		const char* client_name = snd_seq_client_info_get_name(client_info);
+		int dst_client_candidate = snd_seq_client_info_get_client(client_info);
+		snd_seq_port_info_set_client(port_info, dst_client_candidate);
+		snd_seq_port_info_set_port(port_info, -1);
+
+		while (snd_seq_query_next_port(midi_out, port_info) == 0) {
+			unsigned int port_caps = snd_seq_port_info_get_capability(port_info);
+			unsigned int port_type = snd_seq_port_info_get_type(port_info);
+			const int type = SND_SEQ_PORT_TYPE_MIDI_GENERIC;
+			const int cap = SND_SEQ_PORT_CAP_WRITE | SND_SEQ_PORT_CAP_SUBS_WRITE;
+
+			if ((port_type & type) == type && (port_caps & cap) == cap)	{
+				// This is a suitable client
+				dst_client = dst_client_candidate;
+				dst_client_name = client_name;
+				dst_port = snd_seq_port_info_get_port(port_info);
+				dst_port_name = snd_seq_port_info_get_name(port_info);
+				candidate_found = true;
+			}
+		}
+	}
+
+	if (!candidate_found) {
+		Output::Debug("ALSA: No suitable client found");
```

Comment:
Because ALSA is not just being used for the sequencer on Linux internally, it could be useful to tell this is about the sequencer for these too.